python/xorbits/deploy/kubernetes/tests/test_kubernetes.py

```diff
@@ -272,3 +276,78 @@ async def test_request_workers_insufficient():
         cluster_api = WebClusterAPI(address=cluster_client.endpoint)
         with pytest.raises(SystemError, match=r".*Insufficient cpu.*"):
             await cluster_api.request_workers(worker_num=1, timeout=30)
+
+
+@pytest.mark.skipif(not kube_available, reason="Cannot run without kubernetes")
+@pytest.mark.skipif(not juicefs_available, reason="Cannot run without juicefs")
+@pytest.mark.asyncio
+async def test_external_storage_juicefs():
+    redis_ip = sp.getoutput(
+        "echo $(kubectl get po redis -o wide) | grep -o '[0-9]*\\.[0-9]*\\.[0-9]*\\.[0-9]*'"
+    )
+    with _start_kube_cluster(
+        supervisor_cpu=0.1,
+        supervisor_mem="1G",
+        worker_num=1,
+        worker_cpu=0.1,
+        worker_mem="1G",
+        external_storage="juicefs",
+        metadata_url="redis://" + redis_ip + ":6379/1",
+        use_local_image=True,
+    ) as cluster_client:
+        import xorbits.pandas as pd
+
+        print(pd.DataFrame({"col": [1, 2, 3]}).sum())
+
+        from kubernetes.stream import stream
+
+        api_client = k8s.config.new_client_from_config()
+        kube_api = k8s.client.CoreV1Api(api_client)
+        pods_name_list = subprocess.getoutput(
+            "kubectl get pods -o name --no-headers=true -n {ns}".format(
+                ns=cluster_client.namespace
+            )
+        ).split("\n")
+        pods_name_list = list(map(lambda x: x[x.index("/") + 1 :], pods_name_list))
+        for pod in pods_name_list:
+            exec_cmd = ["/bin/sh", "-c", "ls /data"]
+            resp = stream(
+                kube_api.connect_get_namespaced_pod_exec,
+                name=pod,
+                namespace=cluster_client.namespace,
+                command=exec_cmd,
+                stderr=True,
+                stdin=False,
+                stdout=True,
+                tty=False,
+            )
+            assert resp != ""
```

Comment:
What's the name of this file ? You can assert that if you know the name. If not, you can assert number of files also.